misc/build.gradle.kts

```diff
@@ -36,8 +36,8 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_17
-        targetCompatibility = JavaVersion.VERSION_17
+        sourceCompatibility = JavaVersion.VERSION_21
```

Comment:
If this isn't a necessity for the PR I would remove this change, we don't know for sure that there's no snippets in the codebase that have dependencies of lower java compat versions.